### DISCUSSION TITLE
[Feedback]Emby or Plex Containers do not work

### DISCUSSION THREAD

**Discussion by bmentink** [IceWhaleTech/CasaOS#1071] [Q&A]
### Description

I tried installing Emby, when I opened it the webpage (8096) briefly opened, then there was a message that there was no app. Plex was the same on 32400. I installed both containers with default settings .. The only docker container I have installed successfully at the moment is FileBrowser.

### Additional Information

	 - OS: linux
	 - CasaOS Version: 0.4.3
	 - Disk Total: 27712 
	 - Disk Used: 8634 
	 - System Info: [Debian GNU/Linux 11   \l Bit:64 Mem:3763M Core:4 Linux CasaOS 5.10.0-22-amd64 #1 SMP Debian 5.10.178-3 (2023-04-22) x86_64 GNU/Linux]
	 - Remote Version: 0.4.3
	 - Browser: Chrome 
	 - Version: 109.0.0.0

**Comment by bmentink** [comment]
Some info from the logs:
```
Info App: File /system/plugins/OMDb.dll has version 1.0.18.0
Info App: Copying plugin from /system/plugins/OMDb.dll to /config/plugins/OMDb.dll
Error App: Error getting version number from /config/plugins/Emby.Webhooks.dll
	*** Error Report ***
	Version: 4.7.11.0
	Command line: /system/EmbyServer.dll -programdata /config -ffdetect /bin/ffdetect -ffmpeg /bin/ffmpeg -ffprobe /bin/ffprobe -restartexitcode 3
	Operating system: Linux version 5.10.0-22-amd64 (debian-kernel@lists.debian.org) (gcc-10 (Debian 10.2.1-6) 10.2.1 20210110, GNU ld (GNU Binutils for Debian) 2.35.2) #1
	Framework: .NET 6.0.8
	OS/Process: x64/x64
	Runtime: system/System.Private.CoreLib.dll
	Processor count: 4
	Data path: /config
	Application path: /system
	System.ArgumentNullException: System.ArgumentNullException: Value cannot be null. (Parameter 'input')
	   at System.Version.Parse(String input)
	   at Emby.Server.Implementations.ApplicationHost.GetDllVersion(String path)
	Source: System.Private.CoreLib
	TargetSite: System.Version Parse(System.String)

```

ALso errors like this:
```
Info App: File /system/plugins/EmbyGuideData.dll has version 1.0.10.0
Info App: Copying plugin from /system/plugins/EmbyGuideData.dll to /config/plugins/EmbyGuideData.dll
Info App: File /config/plugins/StudioImages.dll has version 1.0.3.0
Info App: File /system/plugins/StudioImages.dll has version 1.0.3.0
Error App: Error getting version number from /config/plugins/Tvdb.dll
	*** Error Report ***
	Version: 4.7.11.0
	Command line: /system/EmbyServer.dll -programdata /config -ffdetect /bin/ffdetect -ffmpeg /bin/ffmpeg -ffprobe /bin/ffprobe -restartexitcode 3
	Operating system: Linux version 5.10.0-22-amd64 (debian-kernel@lists.debian.org) (gcc-10 (Debian 10.2.1-6) 10.2.1 20210110, GNU ld (GNU Binutils for Debian) 2.35.2) #1
	Framework: .NET 6.0.8
	OS/Process: x64/x64
	Runtime: system/System.Private.CoreLib.dll
	Processor count: 4
	Data path: /config
	Application path: /system
	System.ArgumentNullException: System.ArgumentNullException: Value cannot be null. (Parameter 'input')
	   at System.Version.Parse(String input)
	   at Emby.Server.Implementations.ApplicationHost.GetDllVersion(String path)
	Source: System.Private.CoreLib
	TargetSite: System.Version Parse(System.String)
```

and ...
```
2023-05-04 01:55:31.453 Info SqliteUserRepository: Opening sqlite connection to /config/data/users.db
2023-05-04 01:55:31.533 Info SqliteUserRepository: Default journal_mode for /config/data/users.db is delete
2023-05-04 01:55:31.741 Error Main: Error in appHost.Init
        *** Error Report ***
        Version: 4.7.11.0
        Command line: /system/EmbyServer.dll -programdata /config -ffdetect /bin/ffdetect -ffmpeg /bin/ffmpeg -ffprobe /bin/ffprobe -restartexitcode 3
        Operating system: Linux version 5.10.0-22-amd64 (debian-kernel@lists.debian.org) (gcc-10 (Debian 10.2.1-6) 10.2.1 20210110, GNU ld (GNU Binutils for Debian) 2.35.2) #1
        Framework: .NET 6.0.8
        OS/Process: x64/x64
        Runtime: system/System.Private.CoreLib.dll
        Processor count: 4
        Data path: /config
        Application path: /system
        SQLitePCL.pretty.SQLiteException: Error: cannot rollback - no transaction is active
        SQLitePCL.pretty.SQLiteException: Exception of type 'SQLitePCL.pretty.SQLiteException' was thrown.
           at SQLitePCL.pretty.SQLiteException.CheckOk(sqlite3 db, Int32 rc)
           at SQLitePCL.pretty.SQLiteException.CheckOk(sqlite3_stmt stmt, Int32 rc)
           at SQLitePCL.pretty.StatementImpl.MoveNext()
           at SQLitePCL.pretty.DatabaseConnection.Execute(IDatabaseConnection This, ReadOnlySpan`1 sqlUtf8)
           at SQLitePCL.pretty.DatabaseConnection.RollbackTransaction(IDatabaseConnection This)
           at Emby.Sqlite.BaseSqliteRepository.GetTableNames(IDatabaseConnection db)
           at Emby.Server.Implementations.Data.SqliteUserRepository.Initialize()
           at Emby.Server.Implementations.ApplicationHost.GetUserRepository()
           at Emby.Server.Implementations.ApplicationHost.RegisterResources()
           at Emby.Server.Implementations.ApplicationHost.Init()
           at EmbyServer.HostedService.StartAsync(CancellationToken cancellationToken)
        Source: SQLitePCL.pretty
        TargetSite: Void CheckOk(SQLitePCLEx.sqlite3, Int32)
```

**Comment by bmentink** [comment]
This is a "Bug" not feedback ..

**Comment by raller1028** [comment]
Emby and Plex take a while to start up properly after installation. Can you try to make sure the app is working 2 minutes after the installation is complete?

**Comment by bmentink** [comment]
No, even after 10 minutes, there is no web server available at the 8096 port.

**Comment by bmentink** [comment]
Plex and Jellyfin behave the same .... however FileBrowser, Dokuwiki and Syncthing work fine ...
Have you actually checked these containers work before adding them to the Apps? Or, are you relying on your users to "test" them for you, if the later, that is not good ..

**Comment by bmentink** [comment]
@raller1028  Any suggestions?  Can you offer suggestions to debug this?

**Comment by tigerinus** [comment]
@bmentink - would you be OK to run Emby and Plex directly from docker CLI (without CasaOS)?

Just want to rule out the possibility of any bug in CasaOS.

Also can you share logs from Plex?

**Comment by bmentink** [comment]
Ok, will try soon

**Comment by bmentink** [comment]
Logs from Plex:
```
[migrations] started
[migrations] no migrations found
───────────────────────────────────────

      ██╗     ███████╗██╗ ██████╗ 
      ██║     ██╔════╝██║██╔═══██╗
      ██║     ███████╗██║██║   ██║
      ██║     ╚════██║██║██║   ██║
      ███████╗███████║██║╚██████╔╝
      ╚══════╝╚══════╝╚═╝ ╚═════╝ 

   Brought to you by linuxserver.io
───────────────────────────────────────

To support LSIO projects visit:
https://www.linuxserver.io/donate/

───────────────────────────────────────
GID/UID
───────────────────────────────────────

User UID:    1000
User GID:    1000
───────────────────────────────────────

**** Server is unclaimed, but no claim token has been set ****
Docker is used for versioning skip update check
[custom-init] No custom files found, skipping...
Starting Plex Media Server. . . (you can ignore the libusb_init error)
Error: Unable to set up server: sqlite3_statement_backend::prepare: disk I/O error for SQL: PRAGMA cache_size=2048 (N4soci10soci_errorE)
Starting Plex Media Server. . . (you can ignore the libusb_init error)
Error: Unable to set up server: sqlite3_statement_backend::prepare: disk I/O error for SQL: PRAGMA cache_size=2048 (N4soci10soci_errorE)
Starting Plex Media Server. . . (you can ignore the libusb_init error)
Error: Unable to set up server: sqlite3_statement_backend::prepare: disk I/O error for SQL: PRAGMA cache_size=2048 (N4soci10soci_errorE)
Starting Plex Media Server. . . (you can ignore the libusb_init error)
Error: Unable to set up server: sqlite3_statement_backend::prepare: disk I/O error for SQL: PRAGMA cache_size=2048 (N4soci10soci_errorE)
Starting Plex Media Server. . . (you can ignore the libusb_init error)
Error: Unable to set up server: sqlite3_statement_backend::prepare: disk I/O error for SQL: PRAGMA cache_size=2048 (N4soci10soci_errorE)
Starting Plex Media Server. . . (you can ignore the libusb_init error)
Error: Unable to set up server: sqlite3_statement_backend::prepare: disk I/O error for SQL: PRAGMA cache_size=2048 (N4soci10soci_errorE)
Starting Plex Media Server. . . (you can ignore the libusb_init error)
Error: Unable to set up server: sqlite3_statement_backend::prepare: disk I/O error for SQL: PRAGMA cache_size=2048 (N4soci10soci_errorE)
Starting Plex Media Server. . . (you can ignore the libusb_init error)
Error: Unable to set up server: sqlite3_statement_backend::prepare: disk I/O error for SQL: PRAGMA cache_size=2048 (N4soci10soci_errorE)
Starting Plex Media Server. . . (you can ignore the libusb_init error)
Error: Unable to set up server: sqlite3_statement_backend::prepare: disk I/O error for SQL: PRAGMA cache_size=2048 (N4soci10soci_errorE)
Starting Plex Media Server. . . (you can ignore the libusb_init error)
Error: Unable to set up server: sqlite3_statement_backend::prepare: disk I/O error for SQL: PRAGMA cache_size=2048 (N4soci10soci_errorE)
Starting Plex Media Server. . . (you can ignore the libusb_init error)
Error: Unable to set up server: sqlite3_statement_backend::prepare: disk I/O error for SQL: PRAGMA cache_size=2048 (N4soci10soci_errorE)
```

**Comment by bmentink** [comment]
Seems in all cases to be File IO errors ..

**Comment by tigerinus** [comment]
> Seems in all cases to be File IO errors ..

Can we preliminarily say this is irrelevant to CasaOS per se?

**Comment by bmentink** [comment]
@tigerinus 
My setup is that the media storage is on a USB stick pool,(two disks) they are formated in XFS ..
/DATA now is a symbol link to /mnt/pool/DATA ... that should not cause this problem though, right? Other Docker containers use this pool just fine ..

**Reply by tigerinus** [reply]
@bmentink - Did you manually symbol link `/DATA` to `/mnt/pool/DATA` or did CasaOS do that?

Did you enable MergerFS?

**Reply by tigerinus** [reply]
filesystem permission handling is tricky - thus CasaOS keep `/DATA` at root. We understand that it can grow over time and use up all spaces eventually, thus we introduced MergerFS feature to allow *merging* extra harddrives to `/DATA`.

re-symlinking `/DATA` is not something we tested. Permission can be wrong at filesystem level, `/DATA` folder level, etc. It's too broad for us to cover in our testing, thus can be quite fragile if you symlink it.

**Reply by bmentink** [reply]
I manually did a `ln -s /mnt/pool/DATA /DATA`
I did not use the GUI MergerFS tools, rather I did it manually on the CLI .... 

Can you offer suggestions on how to fix this?

**Reply by bmentink** [reply]
Just a followup note: My use case is to use the Zimaboard as a NAS system, so in keeping with that, the data pool does NOT contain the operating system, so that the data is completely seperate from the OS.

That is why I did not want to have /DATA on both the OS and the data pool.  I have used MergerFS to create my pool and SnapRaid to back it up.

**Reply by tigerinus** [reply]
I see.

Do you mind provide a minimum number of steps so we can reproduce this issue on our Zimaboard?

**Reply by bmentink** [reply]
> I see.
> 
> Do you mind provide a minimum number of steps so we can reproduce this issue on our Zimaboard?

Well roughly I did the following:

1.  Manually mounted my two disks (just parity disk and data disk .. will add more data later)
2. Added mounts in /etc/fstab
3. Added MergerFS mount in fstab: LIke this:

My data drives
/dev/disk/by-uuid/ae9660ef-25b9-4575-af36-4f0911295bd1 /mnt/parity1       xfs  defaults 0 2
/dev/disk/by-uuid/7e350887-1308-4c3e-97c4-9711043cf166 /mnt/disk1         xfs  defaults 0 2
#MergerFS
/mnt/disk*    /mnt/pool    fuse.mergerfs    direct_io,defaults,allow_other,minfreespace=50G,fsname=mergerfs 0 0

4.  Deleted original /DATA
5. Created a /DATA directory in the root of /mnt/pool
6. Created a soft link to the root of the OS with "ln -s /mnt/pool/DATA /DATA
7.  All my Docker data now goes into the pool ..
8.  Set up SnapRaid for the two drives.

Hope That helps @tigerinus

**Comment by tigerinus** [comment]
That's right.

Try using some filesystem that's not XFS and not on USB stick just to keep ruling out different factors.

**Comment by bmentink** [comment]
@tigerinus  Hmm, I reinstalled Emby, but pointed the media to the internal SSD (ext4)  of my ZimaBoard, and it works fine.

Why won't it work on my USB SnapRaid pool?  All my data (>3TB) is on there ... I am not going to change my XFS file system and reinstall 3TB of data ... so there must be a better solution ..

**Reply by tigerinus** [reply]
Since this no longer an issue in CasaOS, converted to discussion.

We can continue figuring out the reason here.

**Comment by bmentink** [comment]
If I point EMBY /config at /mnt/pool/DATA/AppData/Emby/config, it seems to have issues running it's DLL files from that location.

However, if I point the /media/movies  at /mnt/pool/DATA/Media/Movies, I can see all my movies and they all have correct permissions and user is abc:abc 

So why is emby having issues with the config location ?

**Comment by bmentink** [comment]
@tigerinus 
So I can run emby fine as long as I have my /config file on the Internal SSD and all media can be on my XFS pool. It's not ideal, but it will do for now. 

My big concern is that the /config file will grow with all the metatdata from Emby and could blow out the storage on my SSD.

I hope this issue can be looked at. I still think this is a BUG
